src/query/expression/src/function.rs

```diff
@@ -436,10 +436,19 @@ where F: Fn(&[ValueRef<AnyType>], &mut EvalContext) -> Value<AnyType> {
             }
         }
         let results = f(&nonull_args, ctx);
-        let bitmap = bitmap.unwrap_or_else(|| constant_bitmap(true, len));
+        let mut bitmap = bitmap.unwrap_or_else(|| constant_bitmap(true, len));
+        let error_bitmap: Bitmap = ctx
+            .errors
+            .as_ref()
+            .map(|(b, _)| b.clone())
+            .unwrap_or_else(|| constant_bitmap(true, len))
+            .into();
+        ctx.errors = None;
+        bitmap = bitmap & &error_bitmap;
+
         match results {
             Value::Scalar(s) => {
-                if bitmap.get(0) {
+                if bitmap.get(0) && ctx.errors.is_none() {
```

Comment:
do not  need to check `ctx.errors` here.

we should update `ctx.errors` by bitmap.

If bitmap is null at some row but it's an error there, the error should be reset.

Evaluator will check `ctx.error` and render it.